lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1784,4 +1786,86 @@ public function validate_user_can_query_private_statuses( $value, $request, $par
 		return new WP_Error( 'rest_forbidden_status', __( 'Status is forbidden' ), array( 'status' => rest_authorization_required_code() ) );
 	}
 
+	/**
+	 * Validate filter param.
+	 *
+	 * This function can be used to validate query params passed by the filter request param.
+	 *
+	 * @param  mixed           $value     Value of the parameter.
+	 * @param  WP_REST_Request $request   Request body.
+	 * @param  string          $parameter Name of the parameter. Should be tax_query.
+	 * @return WP_Error|boolean
+	 */
+	public function rest_validate_query_filter( $values, $request, $parameter ) {
+		// Returns true for validation by default if $values is empty or if a filter param is not being validated.
+		$is_valid = true;
+		if ( ! empty( $values ) ) {
+			foreach ( $values as $filter_param => $value ) {
+				switch ( $filter_param ) {
+					case 'tax_query' :
+						$is_valid = $this->rest_validate_tax_query( $value, $request, $filter_param );
+						// If an error is found exit early and it will throw an error for request->has_valid_params();
+						if ( is_wp_error( $is_valid ) ) {
+							return $is_valid;
+						}
+						continue;
+					default :
+						continue;
```

Comment:
Yeah it was originally an entire validation of tax_query, but now it handles only one case.  If this ever wanted to be expanded though it should go back to switch.  I can make a modification to it sometime soon.